cmd/retry.go

```diff
@@ -91,16 +91,28 @@ func newRetryTimer(unit time.Duration, cap time.Duration, jitter float64, doneCh
 	go func() {
 		defer close(attemptCh)
 		nextBackoff := 0
+		// Channel used to signal after the expiry of backoff wait seconds.
+		var backoffWait <-chan time.Time
 		for {
+			// only done channel and the attempt channel are enabled initially.
+			// done channel is always responsive.
 			select {
 			// Attempts starts.
 			case attemptCh <- nextBackoff:
 				nextBackoff++
+				// disable the attempt channel till next backoff time.
+				attemptCh = nil
+				// enable the channel to notify after nextbackoff number of seconds.
+				backoffWait = time.After(exponentialBackoffWait(nextBackoff))
 			case <-doneCh:
 				// Stop the routine.
 				return
+			case <-backoffWait:
+				// disable till next backoff time is set in first case.
+				backoffWait = nil
+				// enable attempt channel.
+				attemptCh = make(chan int)
```

Comment:
Right. Should see how else the `done` channel be made responsive without the routine going to sleep.